azure-quantum/tests/unit/test_job_results.py

```diff
@@ -61,8 +61,201 @@ def test_job_for_microsoft_quantum_results_v1_invalid_histogram_returns_raw_resu
+        self.assertTrue(len(job_results) == 4)
+        self.assertEqual(job_results[0], (0,))
+        self.assertEqual(job_results[1], (1,))
+        self.assertEqual(job_results[2], (1,))
+        self.assertEqual(job_results[3], (0,))
+    
+    def test_job_for_microsoft_quantum_results_shots_batch_v2_success(self):
+        job_results = self._get_job_results_shots("microsoft.quantum-results.v2","{\"DataFormat\": \"microsoft.quantum-results.v2\", \"Results\": [{\"TotalCount\": 4, \"Histogram\": [{\"Outcome\": [0], \"Display\": \"[0]\", \"Count\": 2}, {\"Outcome\": [1], \"Display\": \"[1]\", \"Count\": 2}], \"Shots\": [[0], [1], [1], [0]]}, {\"TotalCount\": 4, \"Histogram\": [{\"Outcome\": [0], \"Display\": \"[0]\", \"Count\": 2}, {\"Outcome\": [1], \"Display\": \"[1]\", \"Count\": 2}], \"Shots\": [[0], [1], [1], [0]]}, {\"TotalCount\": 4, \"Histogram\": [{\"Outcome\": [0], \"Display\": \"[0]\", \"Count\": 2}, {\"Outcome\": [1], \"Display\": \"[1]\", \"Count\": 2}], \"Shots\": [[0], [1], [1], [0]]}]}")
+        self.assertTrue(len(job_results) == 3)
+        for i in range(3):
+            self.assertTrue(len(job_results[i]) == 4)
+            self.assertEqual(job_results[i][0], (0,))
+            self.assertEqual(job_results[i][1], (1,))
+            self.assertEqual(job_results[i][2], (1,))
+            self.assertEqual(job_results[i][3], (0,))
+
+    def test_job_for_microsoft_quantum_results_histogram_v2_tuple_success(self):
+        output = '''{
+            \"DataFormat\": \"microsoft.quantum-results.v2\",
+            \"Results\": [
+                {
+                \"TotalCount\": 23,
+                \"Histogram\": [
+                    {
+                    \"Outcome\": {
+                        \"Item1\": [
+                        1,
+                        0
+                        ],
+                        \"Item2\": {
+                        \"Item1\": -2.71,
+                        \"Item2\": 67
+                        }
+                    },
+                    \"Display\": \"([1, 0], (-2.71, 67))\",
+                    \"Count\": 1
+                    },
+                    {
+                    \"Outcome\": [1, 0],
+                    \"Display\": \"[1, 0]\",
+                    \"Count\": 1
+                    },
+                    {
+                    \"Outcome\": [1],
+                    \"Display\": \"[1]\",
+                    \"Count\": 1
+                    }
+                ],
+                \"Shots\": [
+                    {
+                    \"Item1\": [
+                        1,
+                        0
+                    ],
+                    \"Item2\": {
```

Comment:
nit: could you please also add one more Shots object like below to cover for additional case
```json
"Item3": [ 
   { 
      "Item1": 6, 
      "Item2": true 
   }, 
   { 
       "Item1": 12, 
       "Item2": false 
   } 
]
```